### PR TITLE
Ransack Implementation

### PR DESCRIPTION
**Pull request by MunazzaZafarAlvi**

### CODE REVIEW

File: app/controllers/home_controller.rb [hunk 1]
```diff
@@ -1,3 +1,8 @@
 class HomeController < ApplicationController
   before_action :authenticate_customer!
+
+  def index
```
Comment (asadsarfraz2347) [inline app/controllers/home_controller.rb]: Ransack applying on product should not be in products_controller?

Comment (asadsarfraz2347) [reply app/controllers/home_controller.rb]: It exists there too, why we are not using products controller?

File: app/controllers/products_controller.rb [hunk 2]
```diff
@@ -1,13 +1,15 @@
 class ProductsController < ApplicationController
   def index
-    @products = Product.all
-    @ladies_products = Product.where(category: 'ladies') # Adjust the condition as per your schema
-    @gents_products = Product.where(category: 'gents')
+    @q = Product.ransack(params[:q])
+    @products = @q.result(distinct: true) # This line should handle the search
+
+    @ladies_products = @products.where(category: 'ladies') # Filter the searched results for ladies
```
Comment (asadsarfraz2347) [inline app/controllers/products_controller.rb]: What is the purpose of these 2 variables, are we showing both on FE or one at a time?

File: app/controllers/home_controller.rb (hunk 1 above)
Comment (MunazzaZafarAlvi) [reply app/controllers/home_controller.rb]: Asad Bhai, I'm already using products_controller.

File: app/controllers/products_controller.rb (hunk 2 above)
Comment (MunazzaZafarAlvi) [reply app/controllers/products_controller.rb]: Which 2 variables are you talking about? @q and @products are using for ransack.

Comment (MunazzaZafarAlvi) [reply app/controllers/products_controller.rb]: Basically we have the option of all products and then the catagories of ladies and gents so both are showing on home page. (Is this what you're aksing?)

**Pull request merged**